src/components/modules/toolbar/inline.ts

```diff
@@ -393,10 +465,32 @@ export default class InlineToolbar extends Module {
 
     /**
      * Filter buttons if Block Tool pass config like inlineToolbar=['link']
+     * Else filter them according to the default inlineToolbar property.
      */
-    if (Array.isArray(inlineToolbarSettings)) {
+
+    let inlineToolbarOrder;
+
+    if (Array.isArray(inlineToolbarSettingsForTool)) {
+      inlineToolbarOrder = inlineToolbarSettingsForTool;
+    } else if (inlineToolbarSettingsForTool == undefined && Array.isArray(this.config.inlineToolbar)) {
```

Comment:
This case should run if `inlineToolbarSettingsForTool === true` as well as if it is not defined